json-rpc/types/src/views.rs

```diff
@@ -313,13 +313,40 @@ impl From<&Vec<u8>> for BytesView {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct VMStatusView {
+    pub major_status: StatusCode,
+    pub context: String,
+    pub sub_status: Option<u64>,
+}
+
+impl From<&VMStatus> for VMStatusView {
+    fn from(status: &VMStatus) -> Self {
+        let (context, sub_status) = ("".to_string(), None);
+        // Once the VMStatus refactor lands
+        //let context = match status.info {
+        //    None => "".to_string(),
+        //    Some(module_id, status) => {
+        //        let error_context = module_id.map(|x| format!("{}::{}", x.address(), x.name()))
+        //            .unwrap_or_else(|| "script".to_string());
+        //        (error_context, Some(status))
+        //    }
+        //};
+        Self {
+            major_status: status.major_status,
+            context,
+            sub_status,
+        }
+    }
+}
+
 #[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct TransactionView {
     pub version: u64,
     pub transaction: TransactionDataView,
     pub hash: String,
     pub events: Vec<EventView>,
-    pub vm_status: StatusCode,
+    pub vm_status: VMStatusView,
```

Comment:
nit: If we just add more items here instead of introducing an new structure, we will make it an non-breaking change.

Reply:
Good point! I would be ok with breaking these out if it would be a significant issue, but I do have a preference for not breaking this out since I think these should be viewed as logically connected.